daemon/api_systems_test.go

```diff
@@ -971,6 +971,7 @@ func (s *systemsSuite) TestSystemsGetSpecificLabelIntegration(c *check.C) {
 						VolumeName: "pc",
 						Type:       "mbr",
 						Role:       "mbr",
+						Offset:     asOffsetPtr(0),
```

Comment:
If we have `partial: [size]` probably the Offset should not be filled, not sure yet what would make more sense. I'd like to fill only things that are unequivocally determined by information in gadget.yaml.

For the API maybe we would need to do what you say, but also you can argue that as the offsets are anyway implicit by gadget data you are still returning the same data. Assuming that you talk about the install API, which currently is used only for subiquity, hopefully we can coordinate to avoid regressions in case there were any.

Reply:
I've been thinking more about this and I think we need this change especially in relation to the `min-size` field. When a partition has `min-size`, it implies that all partitions below until there is one with a fixed offset have undetermined offset. If we apply this change, that will simplify handling this problem: if offset is set in `VolumeStructure` we will know that it is fixed (we do not really care if it was explicitly or implicitly set), otherwise we will need to resolve the offsets with values that will depend on the available space on disk. The resolved offsets will be set in `LaidoutStructure` when calculated.

Reply:
@alfonsosanchezbeato @mvo this made me look and I noticed when we reused the gadget structs in the API we didn't add any omitempty, so I suppose they might have a bunch of "null" in places now? I suppose we need to discuss with foundations if we can clean that up